pkgs/_cfe_macros/test/cfe_test.dart

```diff
@@ -0,0 +1,154 @@
+// Copyright (c) 2024, the Dart project authors. Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+import 'dart:io';
+
+import 'package:front_end/src/macros/macro_injected_impl.dart';
+import 'package:frontend_server/compute_kernel.dart';
+import 'package:macros/macros.dart';
+import 'package:macros/src/executor.dart';
+import 'package:test/test.dart';
+
+void main() {
+  group('CFE with injected macro implementation', () {
+    late File productPlatformDill;
+    late Directory tempDir;
+    late TestMacroPackageConfigs packageConfigs;
+    late TestMacroRunner runner;
+
+    setUp(() {
+      // Set up CFE.
+
+      // TODO(davidmorgan): this dill comes from the Dart SDK running the test,
+      // but `package:frontend_server` and `package:front_end` are used as a
+      // library, so we will see version skew breakage. Find a better way.
+      productPlatformDill = File('${Platform.resolvedExecutable}/../../'
+          'lib/_internal/vm_platform_strong_product.dill');
+      if (!File.fromUri(productPlatformDill.uri).existsSync()) {
+        throw StateError('Failed to find platform dill: $productPlatformDill');
+      }
+      tempDir = Directory.systemTemp.createTempSync('cfe_test');
+
+      // Inject test macro implementation.
+      packageConfigs = TestMacroPackageConfigs();
+      runner = TestMacroRunner();
+      macroImplementation = MacroImplementation(
```

Comment:
Is this a global in the CFE? Kind of hard to understand where the implementation is actually being set when it is done this way.